### DISCUSSION TITLE
Create menu at run time

### DISCUSSION THREAD

**Discussion by 2Skye** [ant-design-blazor/ant-design-blazor#3241] [Q&A]
Good morning
I'm trying to build a run-time menu so that I can use a table that contains an id, a description and possibly a route to be able to call internal windows.
The menu has the possibility of reaching up to a maximum of 4 sub-levels.
I tried several solutions before reaching for help.
I'm getting the doubt that I can't do what I intended.
The above would help me to make the management of the implementation of the menu itself independent.
Can anyone help me fix the problem.
Waiting for your welcome help, I send cordial greetings to the whole community

Luca Perini

**Comment by ElderJames** [comment]
Hello @2Skye , I think you can just use the for-loop to do that.

**Comment by 2Skye** [comment]
Hi EiderJames
Sorry for the delay in replying but I had problems and thanks for your reply
I tried with a for loop and it works but it creates a menu with a submenu and a select menu
Ex: submenu => menu
while I need to have a
submenu => submenu => menu but this doesn't do it
However, the menu that I find in the various submenus created always and only point to a page that is in the href of the last menu used
The menu layout I need would be:
menu
submenu submenu menu
menu submenu
menu (the values of this menu can be found in the various sub-menus)
If you have an idea, thank you for sharing it with me

Thanks again

Luca
